### ISSUE TITLE
update eslint-plugin-universe to support prettier v3

### ISSUE THREAD

**Issue by rafipiccolo** [expo/expo#23648]
### Minimal reproducible example

https://snack.expo.dev/@m.s/eslint-with-prettier-3

### Summary

eslint-plugin-universe is not using latest eslint-plugin-prettier which result in TypeError: prettier.resolveConfig.sync is not a function

prettier version 3 came and allows to go completely {type: module} in th package .json
now i can name my config file .prettierrc.js
instead of .prettierrc.cjs and use export default ...

https://prettier.io/blog/2023/07/05/3.0.0.html

i can confirm it works.

but now when i run eslint i have an error caused by eslint-plugin-universe, because it uses version eslint-plugin-prettier^4 instead of eslint-plugin-prettier@5 which should use prettier 3.

### Environment

```
expo-env-info 1.0.5 environment info:
    System:
      OS: Linux 5.15 Ubuntu 22.04.2 LTS 22.04.2 LTS (Jammy Jellyfish)
      Shell: 5.1.16 - /bin/bash
    Binaries:
      Node: 20.4.0 - /usr/local/bin/node
      Yarn: 1.22.19 - /usr/local/bin/yarn
      npm: 9.8.1 - /usr/local/bin/npm
    npmPackages:
      babel-preset-expo: ^9.3.0 => 9.3.0 
      expo: ^48.0.4 => 48.0.4 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.71.3 => 0.71.3 
    npmGlobalPackages:
      eas-cli: 3.16.0
      expo-cli: 6.3.9
    Expo Workflow: managed
```

**Comment by Mostafatalaat770** [comment]
Confirmed!

**Comment by grifotv** [comment]
I get this error `TypeError: prettier.resolveConfig.sync is not a function` when using `"eslint-config-universe": "^11.3.0"` and `"prettier": "^3.0.0"`.

When downgrading to `"prettier": "^2.8.8"` it works without issues, so I believe I'm experiencing the same issue.

**Comment by Mostafatalaat770** [comment]
This should be fixed by [#23544](https://github.com/expo/expo/pull/23544)
